src/logworker.cpp

```diff
@@ -110,8 +110,13 @@ namespace g3 {
       _impl._bg->send([this, fatal_message] {_impl.bgFatal(fatal_message); });
    }
 
-   void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
-      auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
+   void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sinkPtr) {
+      std::weak_ptr<g3::internal::SinkWrapper> sinkWeakPtr = sinkPtr;
+      auto bg_addsink_call = [this, sinkWeakPtr] {
+         if (auto sinkPtr = sinkWeakPtr.lock()) {
```

Comment:
Please don't reuse a function member's name. That can lead to hard to find bugs.